packages/grafana-runtime/src/utils/queryResponse.ts

```diff
@@ -93,6 +94,79 @@ export function toDataQueryResponse(res: any): DataQueryResponse {
   return rsp;
 }
 
+/**
+ * Parse the results from /api/ds/query into a DataQueryResponse ordering it by query input RefID's.
+ *
+ * @public
+ */
+export function toDataQueryResponseOrdered(queries: DataQuery[], res: any): DataQueryResponse {
+  const rsp: DataQueryResponse = { data: [], state: LoadingState.Done };
+  if (res.data?.results) {
+    const results: KeyValue = res.data.results;
+    for (const q of queries) {
+      const refId = q.refId;
+      const dr = results[refId] as DataResponse;
+      if (dr) {
```

Comment:
nit: Maybe do a quick exit here if possible too.
```suggestion
      if (!dr) {
        continue;
      }
```

Reply:
Copied the code from https://github.com/grafana/grafana/blob/1d399b89eab6ad276e06cc85bc308ecd2aea2776/packages/grafana-runtime/src/utils/queryResponse.ts#L29 so haven't considered any changes to it, but it makes sense.

Reply:
The question is more does it make sense to have a `toDataQueryResponseOrdered` function in combination with `toDataQueryResponse` in the grafana/runtime package?

Reply:
Hmm, yes, that is a good question. Something feels a bit odd. Why should I as a consumer have to think about order? 
Shouldn't this be the default behavior handled by the framework in `toDataQueryResponse`, but there are probably valid use cases I'm not understanding, like streaming?

Reply:
Discussed with Ryan and he suggested to update `toDataQueryResponse` to take an optional queries argument rather than adding a new method `toDataQueryResponseOrdered`.

Reply:
Fixed suggestion in updated `toDataQueryResponse`